### ISSUE TITLE
Adding allSettled function as in Q library

### ISSUE THREAD

**Issue by fahad86** [suguru03/aigle#101]
Q library implementation:

```
/**
 * Turns an array of promises into a promise for an array of their states (as
 * returned by `inspect`) when they have all settled.
 * @param {Array[Any*]} values an array (or promise for an array) of values (or
 * promises for values)
 * @returns {Array[State]} an array of states for the respective values.
 */
Promise.prototype.allSettled = function () {
    return this.then(function (promises) {
        return all(array_map(promises, function (promise) {
            promise = Q(promise);
            function regardless() {
                return promise.inspect();
            }
            return promise.then(regardless, regardless);
        }));
    });
};
```

**Comment by suguru03** [comment]
Thanks for the suggestion! It looks really useful! 😄

**Comment by suguru03** [comment]
Published on v1.14.0! 🙂

**Comment by fahad86** [comment]
> Published on v1.14.0! 🙂

Can you please commit the compiled version of aigle-es5.js to the repo? Right now, it's just empty.

**Comment by suguru03** [comment]
Oh, I didn't realize it, I will fix it soon!
